lib/Models/TableCatalogItem.js

```diff
@@ -208,6 +208,40 @@ var TableCatalogItem = function(terria, url, options) {
             this._rectangle = rect;
         }
     });
+
+    overrideProperty(this, 'dataUrl', {
+        // item.dataUrl returns a URI which downloads the table as a csv.
+        get: function() {
+            if (defined(this._dataUrl)) {
+                return this._dataUrl;
+            }
+            // If the file only exists locally, do not recreate it as a data URI.
+            if (this._dataUrlType === 'local') {
+                return '';
+            }
+            if (defined(this._tableStructure)) {
+                return this._tableStructure.toDataUri();
+            }
+        },
+        set: function(value) {
+            this._dataUrl = value;
+        }
+    });
+
+    overrideProperty(this, 'dataUrlType', {
+        get: function() {
+            if (defined(this._dataUrlType)) {
+                return this._dataUrlType;
+            }
+            if (defined(this._tableStructure)) {
+                return 'data-uri';
```

Comment:
good point. Although it's also useful for it to know that the url is a data-uri, not a link to an external csv file.  So perhaps `csv-data-uri`...